### PR TITLE
ATSAM4N support

### PR DESCRIPTION
**Pull request by john-terrell**
This PR added ATSAM4N support.   The clock setup is similar to how the ATSAM4E setup works.

Tests:
* Boots the "Blinky" sample in the sam_xplained repo on the sam4n_xplained_pro development board.

### CODE REVIEW

**Comment by john-terrell** [comment]
I probably broke the GitHub workflow.    I looked up how to do logical OR in the conditional but I probably got it wrong.

File: src/clock.rs [hunk 1]
```diff
@@ -79,6 +77,19 @@ fn setup_main_clock(pmc: &PMC, main_clock: MainClock) -> Hertz {
             // 0 = no prescaling
             0
         }
+        #[cfg(feature = "atsam4n")]
+        MainClock::RcOscillator4Mhz => {
+            switch_main_clock_to_fast_rc_4mhz(pmc);
+
+            // Set up the PLL for 100Mhz operation (4Mhz RC * (25 / 1) = 100Mhz)
+            let multiplier: u16 = 25;
```
Comment (haata) [inline src/clock.rs:85]: 4 x 50 = 100 MHz

File: src/clock.rs [hunk 2]
```diff
@@ -90,6 +101,19 @@ fn setup_main_clock(pmc: &PMC, main_clock: MainClock) -> Hertz {
             // 0 = no prescaling
             0
         }
+        #[cfg(feature = "atsam4n")]
+        MainClock::RcOscillator8Mhz => {
+            switch_main_clock_to_fast_rc_8mhz(pmc);
+
+            // Set up the PLL for 80Mhz operation (8Mhz RC * (12 / 1) = 96Mhz)
+            let multiplier: u16 = 12;
```
Comment (haata) [inline src/clock.rs:109]: 8 x 25 = 100 MHz

**Review by haata — CHANGES_REQUESTED** [review]

File: src/clock.rs (hunk 2 above)
Comment (haata) [reply src/clock.rs:107]: I guess what I meant to say earlier is that
8 MHz * 25 / 2 = 100 MHz (which is what the SAM4N asf example code does)

**Comment by john-terrell** [comment]
@haata - Going through the ASF code I noticed that the 4Mhz RC Oscillator isn't supported as a PLL input on the 4N.   I've removed that as an option for the 4N configuration..

**Review by haata — APPROVED** [review]

**Pull request merged**